app/javascript/components/map/VMapPASearch.vue

```diff
@@ -1,17 +1,88 @@
 <template>
-  <div>
-    Search
+  <div class="v-map-pa-search">
+    <input
+      ref="input"
+      class="v-map-pa-search__input"
+      type="text"
+      :placeholder="type.placeholder"
+      v-model="query"
+      @input="onInput"
+      @keyup.enter.prevent="onEnter"
+    />
+    <div class="v-map-pa-search__search_icon" @click="onIconClick" />
   </div>
 </template>
 
 <script>
+import axios from "axios";
+
 export default {
-  name: 'VMapPASearch',
+  name: "VMapPASearch",
+
+  model: {
+    prop: "query",
+    event: "search"
+  },
 
-  data () {
+  data() {
     return {
+      query: "",
+      autocompleteResults: [],
+    };
+  },
+
+  props: {
+    type: {
+      type: Object,
+      required: true,
+      validator: type => {
+        return type.hasOwnProperty('id') &&
+          type.hasOwnProperty('placeholder')
+      }
     }
   },
 
-}
+  computed: {
+    hasValidQuery() {
+      return this.query && this.query.length > 2;
+    }
+  },
+
+  methods: {
+    onInput(e) {
+      this.$emit("input", e.target.value);
+    },
+
+    focusInput() {
+      this.$refs.input.focus();
+    },
+
+    onIconClick() {
+      if (this.query) {
+        this.submitSearch();
+      } else {
+        this.focusInput();
+      }
+    },
+
+    onEnter() {
+      this.$emit("search", this.query);
+      this.submitSearch()
+        .then(results => {
+          this.autocompleteResults = results;
+        })
+        .catch(e => {
+          console.error(e);
+          this.autocompleteResults = [];
+        });
+    },
+
+    submitSearch() {
+      return axios.post("/search/autocomplete", {
```

Comment:
I think you may need to add a header to axios (see Stacy's axios mixin)

Reply:
Can we not just bind axios on the window object or on the Vue prototype e.g. Vue.prototype.$http = axios so we can use it from any component as this.$http without the need to re-import? We can set all the constraints on axios via something like axios.defaults.headers - it then saves us having to do this every time

Reply:
This makes sense to me, would be good if you could do that. If you do this, can you leave the map request helpers as is? They use different axios config.

Reply:
If there's a slightly different config, I see what you mean. I can implement this.$http more generally if we agree, I'll have a look at the axios mixin, I'm not yet clear on what's needed